src/plugins/index_pattern_field_editor/public/components/field_editor/field_editor.tsx

```diff
@@ -221,6 +233,9 @@ const FieldEditorComponent = ({
         </EuiFlexItem>
       </EuiFlexGroup>
 
+      {(nameHasChanged || typeHasChanged) && (
+        <EuiFormErrorText data-test-subj="changeWarning">{changeWarning}</EuiFormErrorText>
```

Comment:
I'm curious what @ryankeairns thinks - I agree that its not the best to conflate designs for errors and warnings, but its also odd to have a warning thats larger than an error.

Reply:
@sebelga @ryankeairns is out on vacation for a week. I think we should move forward with this and circle back later.

Reply:
Maybe @mdefazio could give us his opinion? Basically between the warning message in the screenshot here and the one in the PR description. Thanks!

Reply:
Here's some quick thoughts:

1. Can we append (or prepend) the warning icon to the form field if it's changed?  and then show that message on hover or click? 
2. Maybe just use the title in the callout instead of title+description.  This will shrink the height of the callout and maybe find a middle ground between our concerns.
3. Perhaps a simpler option would be to show a confirmation dialog with this message when saving. If we do option 1, I think we would still need this approach too.

Happy to work through some more options as well as these are just some that come to mind.

Reply:
I'm a bit confused about the problem we're trying to solve with the current design. @sebelga did you want something bigger since the change has the potential to break things?

A confirm dialog has already been implemented - https://github.com/elastic/kibana/pull/95237

Reply:
My concern with the original solution is that we are displaying it as if it was an **error** (red). And it is not, there is no error. What we want is display a **warning** to the user (yellow) to make sure he knows what he is doing. That's why I thought a warning callout was better.

Maybe _not_ showing the title in the callout (so removing the icon and "Name/type change" would be enough? Not sure that's possible though, but option 2 from @mdefazio could work.

Reply:
To me, the warning callout is too loud. We follow it up with confirm dialog with the same content. Do we really need both? 

I really don't like just showing the title of the warning as then we're failing to explain _why_ we're showing it.

Reply:
I don't want to block on this, just sharing the pattern we've been applying in almost all of our forms. When there is something the user needs to be aware of in a form we either user the yellow or the blue callout. Even if it might look "big", if we consider that it is an important information for him to know we use the EuiCallout as they are meant for that.

If we consider it is not _that_ important in the form and prefer just to have to confirm modal,  that could be enough. But personally I don't mind having both to inform the user as soon as possible.

But again, happy if we go either way and don't want to block this PR on this. 😊

Reply:
> I really don't like just showing the title of the warning as then we're failing to explain why we're showing it.

I agree with this, and to be clearer, my thought was putting `Changing name or type...` into the title prop of the callout.  I don't think we need the `Name / type change`  

I would prefer seeing either both a callout + modal confirmation or just the modal confirmation.  But I think we should avoid using the help/error text area in the input for warnings.

Reply:
I spoke to the runtime fields ui working group about this and we've decided to go with the single line warning callout with icon.

Reply:
updated description with new screenshot